src/System.Management.Automation/engine/hostifaces/LocalConnection.cs

```diff
@@ -146,32 +146,45 @@ public override PSThreadOptions ThreadOptions
             {
                 lock (this.SyncRoot)
                 {
-                    if (value != _createThreadOptions)
+                    if (value == _createThreadOptions)
                     {
-                        if (this.RunspaceStateInfo.State != RunspaceState.BeforeOpen)
-                        {
-#if CORECLR                 // No ApartmentState.STA Support In CoreCLR
-                            bool allowed = value == PSThreadOptions.ReuseThread;
-#else
-                            // if the runspace is already opened we only allow changing the options if
-                            // the apartment state is MTA and the new value is ReuseThread
-                            bool allowed = (this.ApartmentState == ApartmentState.MTA || this.ApartmentState == ApartmentState.Unknown) // Unknown is the same as MTA
-                                           &&
-                                           value == PSThreadOptions.ReuseThread;
-#endif
+                        return;
+                    }
 
-                            if (!allowed)
-                            {
-                                throw new InvalidOperationException(StringUtil.Format(RunspaceStrings.InvalidThreadOptionsChange));
-                            }
+                    if (this.RunspaceStateInfo.State != RunspaceState.BeforeOpen)
+                    {
+                        if (IsInvalidThreadOptionsConfiguration(value))
+                        {
+                            throw new InvalidOperationException(StringUtil.Format(RunspaceStrings.InvalidThreadOptionsChange));
                         }
-
-                        _createThreadOptions = value;
                     }
+
+                    _createThreadOptions = value;
                 }
             }
         }
 
+        private bool IsInvalidThreadOptionsConfiguration(PSThreadOptions options)
+        {
+            // If the runspace is already opened, we only allow changing options when:
+            //  - The new value is ReuseThread, and
+            //  - The apartment state is MTA
+
+            if (options == PSThreadOptions.ReuseThread)
+            {
+                return false;
+            }
```

Comment:
This doesn't seem right, since it will return false even if `this.ApartmentState` is `STA`.

Reply:
I'm going to simplify this to just check option and apartment state